### ISSUE TITLE
MSVC 2017 and the _WIN32_WINNT flag

### ISSUE THREAD

**Issue by kwilliamszaf** [inexor-game/conan-grpc#17]
Hi

Would it be possible to please include a default value for _WIN32_WINNT in gRPC 1.20.0+ since I currently get the error:

> d:\.conan\d6ea00\1\include\grpc/impl/codegen/port_platform.h(44): fatal error C1189: #error:      "Please compile grpc with _WIN32_WINNT of at least 0x600 (aka Windows Vista)" (compiling source file xxx)

Thanks!

**Comment by Croydon** [comment]
I recently added it for MinGW builds: https://github.com/inexorgame/conan-grpc/pull/14

But I'm wondering now two things:
1) Why isn't this an issue in our Windows CI? Is it added implicitly somewhere along the build tools?
2) Shouldn't this maybe be fixed upstream instead? E.g. https://github.com/grpc/grpc/issues/18928

//cc @SebastianMosiej

**Comment by SebastianMosiej** [comment]
Hi, Please notice that in **conanfily.py** there's statement
`        if self.settings.os == "Windows" and self.settings.compiler == "gcc":`
It was added to allow mentioned flag be set. @kwilliamszaf problem can be easily solved by removing **and self.settings.compiler == "gcc"**.
However, as you mentioned in second point - in gRPC project they already discussing it.

**Comment by Croydon** [comment]
@kwilliamszaf Is this still a problem you run into with the current grpc/1.23.0?

**Comment by AnatoliiShablov** [comment]
Same for 1.29.1, but can you please remove check for gcc compiler. It's fixed in 1.3x. (I don't remember). But for earlier versions we still need it.

**Comment by Croydon** [comment]
Yes, I'm aware. Newer version >= 1.30.0 are still blocked on https://github.com/conan-io/conan-center-index/pull/2037

**Comment by Croydon** [comment]
This recipe is now in the [Conan Center Index](https://github.com/conan-io/conan-center-index/).

Please migrate to the recipe version there. 
If this is still an issue, please create a new issue in the [Conan Center Index issue tracker](https://github.com/conan-io/conan-center-index/issues).

Thanks!

https://conan.io/center/grpc
